### ISSUE TITLE
ValidationError caused by id mismatch

### ISSUE THREAD

**Issue by msakai** [chainer/onnx-chainer#28]
In the following code it seems there are cases where `i.get_variable_or_none()` returns `Variable/Parameter` and go into `else` branch even if it is not a parameter inside a Link nor network input.

https://github.com/chainer/onnx-chainer/blob/8211ad69059376c26c92e3d2c6c59f5a0c07b692/onnx_chainer/export.py#L77-L100

In such cases, consumer of the variable refer it using `id` of `Variable` while producer of the variable refer it using `id` of `VariableNode`. This mismatch causes `onnx.onnx_cpp2py_export.checker.ValidationError` like following:

> onnx.onnx_cpp2py_export.checker.ValidationError: Nodes in a graph must be topologically sorted, however input '112084623768' of node: input: "112084623768" output: "112084576968" op_type: "Reshape" attribute { name: "shape" ints: 32 ints: 51 ints: 16 type: INTS }

**Comment by msakai** [comment]
Following is a program that reproduce the problem.

```python
import numpy as np
import chainer
import chainer.functions as F
import chainer.links as L
import onnx_chainer

class Model(chainer.Chain):
    def __init__(self, out_dim, hidden_dim=16):
        super(Model, self).__init__()
        with self.init_scope():
            self.update_layer = L.GRU(2 * hidden_dim, hidden_dim)

    def __call__(self, h):
        out_h = self.update_layer(h)
        out_h = F.reshape(out_h, (1, 3, 16))
        return out_h

model = Model(out_dim=16, hidden_dim=16)

# Pseudo input
h = np.zeros((3, 32), dtype=np.float32)

onnx_chainer.export(model, h, filename='test.onnx')
```

**Comment by msakai** [comment]
I applied following patch to avoid the problem for my case, although I think it is not a proper fix.

```
diff --git a/onnx_chainer/export.py b/onnx_chainer/export.py
index 62aaeed..d5e2013 100644
--- a/onnx_chainer/export.py
+++ b/onnx_chainer/export.py
@@ -85,10 +85,15 @@ class ONNXExport(chainer.FunctionHook):
                         str(id(i)) not in self.network_inputs:
                     self.network_inputs[str(id(i))] = i
             else:  # It is a parameter inside a Link or network input
-                input_names.append(str(id(var)))
                 if i.creator is None and \
                         not isinstance(var, chainer.Parameter):
+                    input_names.append(str(id(var)))
                     self.network_inputs[str(id(var))] = var
+                else:
+                    if isinstance(var, chainer.Parameter):
+                        input_names.append(str(id(var)))
+                    else:
+                        input_names.append(str(id(i)))
 
         # This is to get corresponding VariableNode id from the output
         # Variable of the network
```

**Comment by disktnk** [comment]
implicit inputs are taken care from v1.3.1, ref #81, I confirmed the blow reproduced example occur error under v1.3.0 and fixed from v1.3.1.

**Comment by msakai** [comment]
Thanks!
